SwiftWisdom/Core/UIKit/UIColor+Hex.swift

```diff
@@ -9,6 +9,7 @@
 import UIKit
 
 public extension UIColor {
+    // TODO: document default behavior for malformed strings (or refactor to failable init)
     convenience init(ip_hex hex: String) {
```

Comment:
Let's just change this to an optional init. I like that better.

Reply:
Ok, sounds good. This means making some adjustments to `ColorDescriptor` as well, since it currently uses this `init` for its `color` computed property.

Reply:
Does that over complicate it? I think it's a better API. Callers should be prepared to provide their own default color instead of having us give it to them.

Reply:
I made some changes to `ColorDescriptor` that return `nil` instead of `fatalError` with bad input for its `init(_ value: String)`. It's not possible to make initializers from string literals failable, so I had to keep the fatal error there.